src/zkevm_specs/evm/execution/memory.py

```diff
@@ -0,0 +1,41 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ..table import RW, CallContextFieldTag
+from ...util import FQ
+
+
+def memory(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    address = instruction.stack_pop()
+
+    is_mload = instruction.is_equal(opcode, Opcode.MLOAD)
+    is_mstore8 = instruction.is_equal(opcode, Opcode.MSTORE8)
+    is_store = FQ(1) - is_mload
+    is_not_mstore8 = FQ(1) - is_mstore8
+
+    value = instruction.stack_push() if is_mload == FQ(1) else instruction.stack_pop()
+
+    src_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    memory_offset = instruction.curr.memory_size
+    next_memory_size, memory_expansion_gas_cost = instruction.memory_expansion(
+        memory_offset, address.expr() + FQ(1) + (is_not_mstore8 * FQ(31))
+    )
+
+    if is_mstore8 == FQ(1):
+        instruction.memory_lookup(RW.Write, address.expr())
+
+    if is_not_mstore8 == FQ(1):
+        for idx in range(32):
+            instruction.memory_lookup(
+                RW.Write if is_store == FQ(1) else RW.Read, address.expr() + idx, src_id
+            )
```

Comment:
```suggestion
            instruction.is_equal(instruction.memory_lookup(
                RW.Write if is_store == FQ(1) else RW.Read, address.expr() + idx, src_id
            ), value.le_bytes[31-idx])
```
Add equality check to value with the lookup